flow-data/src/main/java/com/vaadin/flow/data/provider/Query.java

```diff
@@ -121,6 +121,9 @@ public int getLimit() {
      * Vaadin asks data from the backend in paged manner. This shorthand
      * calculates the page index for backends using paged data access, such as
      * Spring Data repositories.
+     * <p>
+     * If page offset is not evenly divisible with pageSize raise pageSize until
+     * it is. Updates the pageSize value if it has been raised.
```

Comment:
nit: could use `page size` for consistency with `page offset` and the Javadoc of `getPageSize()`. 
```suggestion
     * If page offset is not evenly divisible with page size raise page size until
     * it is. Updates the page size value if it has been raised.
```

Reply:
updated